tests/protocol/test_wire.py

```diff
@@ -1070,96 +1070,98 @@ def test_it_should_return_true_on_success(self):
     def test_it_should_return_false_on_failure(self):
         with mock_wire_protocol(mockwiredata.DATA_FILE) as protocol:
             def http_get_handler(url, *_, **__):
-                if url.endswith('/machine/?comp=goalstate'):
+                if self.is_goal_state_request(url):
                     return HttpError('Exception to fake an error retrieving the goal state')
                 return None
 
-            with mock_http_request(http_get_handler=http_get_handler):
-                self.assertFalse(protocol.client.try_update_goal_state(), "try_update_goal_state should have failed")
+            protocol.set_http_handlers(http_get_handler=http_get_handler)
```

Comment:
i need to start mocking *after* the protocol is fully constructed because i'm failing goal state requests